code/src/components/Details.js

```diff
@@ -0,0 +1,107 @@
+import React, { useState, useEffect } from "react"
+import { useParams, useNavigate } from "react-router-dom"
+import { SINGLE_MOVIE_URL } from "utils/urls"
+import styled from "styled-components"
+
+const Details = () => {
+
+    const [details, setDetails] = useState({})
+
+    //renamed to movieId for clarity. Also, sent it as an argument to the fetch function.
+    const { movieId } = useParams()
+    const navigate = useNavigate()
+
+
+    const onBackButtonClick = () => {
+        navigate(-1)
+    }
+
+
+    useEffect(() => {
+        fetch(SINGLE_MOVIE_URL(movieId))
+            .then(res => res.json())
+            .then(data => setDetails(data))
+    }, [movieId])
+
+    return (
+        <MovieDetailsPage bgImage={`https://image.tmdb.org/t/p/w1280/${details.backdrop_path}`}>
+            <Container>
+                <div><GoBackButton onClick={onBackButtonClick}>← Go back to movies</GoBackButton></div>
+                <MovieDetails>
+                    <MovieImage src={`https://image.tmdb.org/t/p/w342/${details.poster_path}`} alt="poster-img" />
+                    <MovieDescription>
+                        <h1>{details.title}</h1>
+                        <div>{details.vote_average}/10</div>
+                        <p>{details.overview}</p>
+                    </MovieDescription>
+                </MovieDetails>
+            </Container>
+        </MovieDetailsPage>
```

Comment:
When opening console after clicking a movie you get an error message like this: https://image.tmdb.org/t/p/w342/undefined 404

We had the same issue and you need to make an if statement to define if `details` = null (undefined) or not. You can also look in our code if you want to see how it looks.

/ Ida & Simon